pulsar-client/src/main/java/org/apache/pulsar/client/impl/AbstractBatchMessageContainer.java

```diff
@@ -53,6 +53,7 @@ public abstract class AbstractBatchMessageContainer implements BatchMessageConta
     // allocate a new buffer that can hold the entire batch without needing costly reallocations
     protected int maxBatchSize = INITIAL_BATCH_BUFFER_SIZE;
     protected int maxMessagesNum = INITIAL_MESSAGES_NUM;
+    private volatile long nanoTimestamp = 0L;
```

Comment:
```suggestion
    private volatile long firstAddedTimestamp = 0L;
```

Maybe we can change it to `firstAddedTimestamp`.